packages/donor/src/components/basic/RadioGroup/RadioGroup.tsx

```diff
@@ -7,35 +7,40 @@ import {
 import { AnalyticsButtonType } from "@zm-blood-components/common";
 import { reportClick } from "../../../Analytics";
 
-export type RadioOption = {
+export type RadioOption<T> = {
   value: any;
   label: string;
   isDisabled?: boolean;
+  key: string;
 };
 
-type RadioGroupProps = {
+type RadioGroupProps<T> = {
   /** For logging and Analytics */
   name: string;
+  getAnalyticsValue?: (optionValue: T | undefined) => string;
   label?: string;
-  onChange: (value: string) => void;
-  value: string;
+  onChange: (value: T) => void;
+  value: T;
   isDisabled?: boolean;
   className?: string;
-  options: RadioOption[];
+  options: RadioOption<T>[];
 };
 
-export default function RadioGroup({
+export default function RadioGroup<T>({
   name,
+  getAnalyticsValue,
   options,
   label,
   onChange,
   value,
   className,
-}: RadioGroupProps) {
-  const handleChange = (newValue: string) => {
-    onChange(newValue);
+}: RadioGroupProps<T>) {
+  const handleChange = (nextValue: string) => {
+    onChange(options.find((option) => option.key === nextValue)?.value!);
```

Comment:
Are you sure you want to compare keys to values?

Reply:
Fixed